### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by alexkovaceski** [autogluon/autogluon#4859]
**Bug Report Checklist**
Seeing multiple errors appear when running Tabular preset= experimental_quality

**Describe the bug**
Getting the following for information  errors when running tabular with experimental quality:

Train Epoch: [0/200] Train Loss: 3.7328 Validation r2: -0.00:  49%|████▉     | 155/314 [01:21<01:23,  1.90it/s]/home/alex/PycharmProjects/algolotl-py/venv/lib/python3.10/site-packages/torch/nn/modules/loss.py:535: UserWarning:
(_ray_fit pid=133433) 
(_ray_fit pid=133433) Using a target size (torch.Size([512])) that is different to the input size (torch.Size([512, 1])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size.
(_ray_fit pid=133433) 

(_ray_fit pid=131398) /home/alex/PycharmProjects/algolotl-py/venv/lib/python3.10/site-packages/xgboost/core.py:160: UserWarning: [repeated 6x across cluster]
(_ray_fit pid=131398) [18:22:00] WARNING: /workspace/src/common/error_msg.cc:45: `gpu_id` is deprecated since2.0.0, use `device` instead. E.g. device=cpu/cuda/cuda:0 [repeated 2x across cluster]
(_ray_fit pid=131398) [18:22:00] WARNING: /workspace/src/common/error_msg.cc:27: The tree method `gpu_hist` is deprecated since 2.0.0. To use GPU training, set the `device` parameter to CUDA instead. [repeated 3x across cluster]
(_ray_fit pid=131398)     E.g. tree_method = "hist", device = "cuda" [repeated 3x across cluster]

and 

itting model: Transformer_BAG_L5 ... Training model for up to 10495.69s of the 10495.68s of remaining time.
	Fitting 5 child models (S1F1 - S1F5) | Fitting with ParallelLocalFoldFittingStrategy (1.0 workers, per: cpus=1, gpus=1, memory=0.00%)
(_ray_fit pid=133433) /home/alex/PycharmProjects/algolotl-py/venv/lib/python3.10/site-packages/autogluon/tabular/models/tab_transformer/tab_transformer_encoder.py:85: FutureWarning:
(_ray_fit pid=133433) 
(_ray_fit pid=133433) unique with argument that is not not a Series, Index, ExtensionArray, or np.ndarray is deprecated and will raise in a future version.
(_ray_fit pid=133433) 
(_ray_fit pid=133433) /home/alex/PycharmProjects/algolotl-py/venv/lib/python3.10/site-packages/autogluon/tabular/models/tab_transformer/tab_transformer_encoder.py:85: FutureWarning:
(_ray_fit pid=133433) 
(_ray_fit pid=133433) unique with argument that is not not a Series, Index, ExtensionArray, or np.ndarray is deprecated and will raise in a future version.
(_ray_fit pid=133433) 
(_ray_fit pid=131655) /home/alex/PycharmProjects/algolotl-py/venv/lib/python3.10/site-packages/xgboost/core.py:160: UserWarning: [repeated 2x across cluster]
(_ray_fit pid=131655) [18:22:07] WARNING: /workspace/src/common/error_msg.cc:27: The tree method `gpu_hist` is deprecated since 2.0.0. To use GPU training, set the `device` parameter to CUDA instead.
(_ray_fit pid=131655)     E.g. tree_method = "hist", device = "cuda"
(_ray_fit pid=133433) /home/alex/PycharmProjects/algolotl-py/venv/lib/python3.10/site-packages/autogluon/tabular/models/tab_transformer/tab_transformer_encoder.py:85: FutureWarning:
(_ray_fit pid=133433) 
(_ray_fit pid=133433) unique with argument that is not not a Series, Index, ExtensionArray, or np.ndarray is deprecated and will raise in a future version.
(_ray_fit pid=133433) 
(_ray_fit pid=133433) /home/alex/PycharmProjects/algolotl-py/venv/lib/python3.10/site-packages/autogluon/tabular/models/tab_transformer/tab_transformer_encoder.py:85: FutureWarning:
(_ray_fit pid=133433) 
(_ray_fit pid=133433) unique with argument that is not not a Series, Index, ExtensionArray, or np.ndarray is deprecated and will raise in a future version.
(_ray_fit pid=133433) 
(_ray_fit pid=133433) /home/alex/PycharmProjects/algolotl-py/venv/lib/python3.10/site-packages/autogluon/tabular/models/tab_transformer/tab_transformer_encoder.py:85: FutureWarning:
(_ray_fit pid=133433) 
(_ray_fit pid=133433) unique with argument that is not not a Series, Index, ExtensionArray, or np.ndarray is deprecated and will raise in a future version.
(_ray_fit pid=133433) 
(_ray_fit pid=133433) /home/alex/PycharmProjects/algolotl-py/venv/lib/python3.10/site-packages/autogluon/tabular/models/tab_transformer/tab_transformer_encoder.py:85: FutureWarning:
(_ray_fit pid=133433) 
(_ray_fit pid=133433) unique with argument that is not not a Series, Index, ExtensionArray, or np.ndarray is deprecated and will raise in a future version.
(_ray_fit pid=133433) 
(_ray_fit pid=133433) /home/alex/PycharmProjects/algolotl-py/venv/lib/python3.10/site-packages/autogluon/tabular/models/tab_transformer/tab_transformer_encoder.py:85: FutureWarning:
(_ray_fit pid=133433) 

**Expected behavior**
System works without any warnings.

**To Reproduce**

custom_hyperparameters_dict: {"GBM": [{"max_depth": 15, "num_leaves": 459, "learning_rate": 0.29, "feature_fraction": 0.55, "bagging_fraction": 0.73, "min_child_samples": 5, "bagging_freq": 2, "max_bin": 241, "lambda_l1": 7.5, "lambda_l2": 24.8, "bagging_seed": 65}], "XT": {}, "KNN": {}, "CAT": {}, "XGB":{}, "NN_TORCH": {}, "LR": {}, "FASTAI": {}, "TRANSF": {}, "FT_TRANSFORMER": {}, "TABPFN": {}, "TABPFNMIX": {}, "ENS_WEIGHTED": {}, "SIMPLE_ENS_WEIGHTED": {},  "IM_RULEFIT": {}, "IM_GREEDYTREE": {},  "IM_FIGS": {}, "IM_HSTREE": {},  "IM_BOOSTEDRULES": {},  "VW": {},"RF": {"criterion": "entropy"}}

    predictor = TabularPredictor(
        label='target',
        eval_metric=evaluation_metric,
        problem_type='regression',
        path=model_directory).fit(
        hyperparameters=custom_hyperparameters_dict,
        num_bag_folds=number_bag_folds,
        num_stack_levels=number_stack_level,
        # fit_strategy = "parallel", #Exploratory feature in 1.2.  doesn't work with GPU yet.
        train_data=df_data,
        #train_data=df_data_low_importance_features_dropped,
        time_limit=model_time_limit,
        presets=model_presets,
        dynamic_stacking=False,
        save_bag_folds=True,
        num_gpus=number_gpus,
        fit_strategy="parallel",
        memory_limit=50000,
        ag_args_fit={
            "use_bag_holdout": True,
            "ag.max_memory_usage_ratio": 3,
            "num_gpus": number_gpus,
            "refit_full": True,
        }
    )

**Installed Versions**
AG 1.2.  Python 3.10.16

**Comment by celestinoxp** [comment]
I had the impression that there is a problem in the xgboost and pandas versions.... try updating the packages and check if the warnings continue.
If they are gone, you can close this issue..

**Comment by Innixma** [comment]
For the XGBoost warning, this is because we haven't updated the parameter used when running on GPU:

```python
        if num_gpus != 0:
            params["tree_method"] = "gpu_hist"
            if "gpu_id" not in params:
                params["gpu_id"] = 0
        elif "tree_method" not in params:
            params["tree_method"] = "hist"
```

The fix is probably:

```
        if num_gpus != 0:
            if "device" not in params:
                params["device"] = "cuda:0"
        if "tree_method" not in params:
            params["tree_method"] = "hist"

```

I created a PR for the change, but haven't tested it on a GPU instance yet: #4866

As for most of the other warnings, I think they are coming from the fitting of the `TRANSF` model, which is very old and not tested. I've created a tracking issue to deprecate this model: #4867.

For this particular warning, I haven't seen it before, and am unsure which model is the one logging it:

```
Using a target size (torch.Size([512])) that is different to the input size (torch.Size([512, 1])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size.
```

**Comment by alexkovaceski** [comment]
Nick, thanks for getting back to me so quickly.  I became a father again
today, so priorities are elsewhere.

I will get you something in the next few days

Regards
Alex

**Comment by Innixma** [comment]
Congratulations @alexkovaceski! No need to rush :)
